tracer/src/Datadog.Tracer.Native/Datadog.Tracer.Native.DLL.vcxproj

```diff
@@ -205,6 +205,7 @@
       <MultiProcessorCompilation Condition=" '$(ENABLE_MULTIPROCESSOR_COMPILATION)' == '' ">true</MultiProcessorCompilation>
       <AdditionalIncludeDirectories>$(LIB_INCLUDES);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <AdditionalOptions>/FS %(AdditionalOptions)</AdditionalOptions>
+      <PreprocessorDefinitions>_SILENCE_STDEXT_ARR_ITERS_DEPRECATION_WARNING;_SILENCE_ALL_MS_EXT_DEPRECATION_WARNINGS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Comment:
What deprecation warnings are we silencing and why? 😅

Reply:
I tend to agree that these deprecation warnings might be too inclusive but I couldn't find any other workaround despite upgrading the problematic module (fmt/spdlog).
These warnings, though, are only are only blocking the Visual Studio compilation. The Nuke script seems to pass just fine...

What do you think about leaving these deprecation warnings out of this PR and reconsider what we want to do?

Reply:
I've deleted these preprocessor definitions from the `vcxproj` files, let's revisit it later.